pytools/persistent_dict.py

```diff
@@ -280,7 +286,10 @@ def update_for_int(key_hash: Hash, key: int) -> None:
         sz = 8
         while True:
             try:
-                key_hash.update(key.to_bytes(sz, byteorder="little", signed=True))
+                # Must match system byte order so that numpy and this
+                # generate the same string of bytes.
+                # https://github.com/inducer/pytools/issues/259
+                key_hash.update(key.to_bytes(sz, byteorder=sys.byteorder, signed=True))
```

Comment:
Out of curiosity, do we actually need to match the numpy representation for ints? Could we just use something like `str(key)` here (similar to what is used for float and complex)?

Reply:
I guess that'd be better. Could you do a PR?